primitives/core/src/traits.rs

```diff
@@ -16,16 +16,35 @@
 
 //! Shareable Substrate traits.
 
-use crate::{crypto::KeyTypeId, ed25519, sr25519};
+use crate::{
+	crypto::{KeyTypeId, CryptoTypePublicPair},
+	ed25519, sr25519,
+};
 
 use std::{
 	fmt::{Debug, Display},
+	collections::HashSet,
 	panic::UnwindSafe,
 	sync::Arc,
 };
 
 pub use sp_externalities::{Externalities, ExternalitiesExt};
 
+/// BareCryptoStore error
+#[derive(Debug)]
+pub enum Error {
+	/// Public key type is not supported
+	KeyNotSupported,
+	/// Pair not found for public key and KeyTypeId
+	PairNotFound(String),
+	/// Validation error
+	ValidationError(String),
+	/// Keystore unavailable
+	Unavailable,
+	/// Programming errors
```

Comment:
Why is that needed? I think the convention is to rather call this variant `Other`

Reply:
Renamed to `Other`. The use case for this is converting keystore error to `BareKeyStoreError` where keystore has `Io` and `Json` errors that could be mapped to `Other` here.